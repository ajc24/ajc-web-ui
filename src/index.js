/* Export all of the header components */
export { default as Header } from './ui-elements/header/Header';

/* Export all of the icon components */
export { default as SmallDownArrowIcon } from './ui-elements/icons/SmallDownArrowIcon';
export { default as SmallRightArrowIcon } from './ui-elements/icons/SmallRightArrowIcon';
