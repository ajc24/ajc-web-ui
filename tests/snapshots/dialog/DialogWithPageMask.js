import 'jsdom-global/register';
import React from 'react';
import { TestDev } from 'ajc-jest-enzyme';
import { DialogWithPageMask } from '../../../src';

describe('DialogWithPageMask', () => {
  const twoButtonsList = [
    {
      id: 'save-button',
      onClick: () => {},
      title: 'Save',
      type: 'submit',
    },
    {
      id: 'cancel-button',
      onClick: () => {},
      title: 'Cancel',
      type: 'button',
    }
  ];

  describe('Default props and rendering', () => {
    let jsonSnapshot;

    beforeAll(() => {
      jsonSnapshot = TestDev.createSnapshot(
        <div role="region">
          <DialogWithPageMask id="test-page-mask-dialog-id" dialogTitle="Test Page Mask Dialog Title" onClose={() => {}}>
            Test Page Mask Dialog Content.
          </DialogWithPageMask>
        </div>
      );
    });

    it('verifies the snapshot for the component', () => {
      expect(jsonSnapshot).toMatchSnapshot();
    });
  });

  describe('Transferred props and rendering - All properties set', () => {
    let jsonSnapshot;

    beforeAll(() => {
      jsonSnapshot = TestDev.createSnapshot(
        <div role="region">
          <DialogWithPageMask id="test-page-mask-dialog-id" dialogTitle="Test Page Mask Dialog Title" dialogContentAreaColour="grey" colour="red"
            buttonsList={twoButtonsList} onClose={() => {}} isDisplayed={true}>
              Test Page Mask Dialog Content.
          </DialogWithPageMask>
        </div>
      );
    });

    it('verifies the snapshot for the component', () => {
      expect(jsonSnapshot).toMatchSnapshot();
    });
  });
});
