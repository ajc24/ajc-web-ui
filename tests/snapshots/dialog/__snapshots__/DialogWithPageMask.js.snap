// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DialogWithPageMask Default props and rendering verifies the snapshot for the component 1`] = `
<div
  role="region"
/>
`;

exports[`DialogWithPageMask Transferred props and rendering - All properties set verifies the snapshot for the component 1`] = `
<div
  role="region"
>
  <div
    className="ajc-flex-row ajc-flex-justify-center ajc-page-mask ajc-width-4k"
    id="page-mask-test-page-mask-dialog-id-root"
  >
    <div
      className="ajc-flex-row ajc-flex-justify-center ajc-width-full-hd"
    >
      <div
        className="ajc-flex-row ajc-flex-justify-center ajc-width-hd"
      >
        <div
          className="ajc-flex-column ajc-flex-justify-start ajc-spacing-top-large ajc-spacing-bottom-large ajc-width-hd-with-padding ajc-text-white ajc-font-default"
          id="page-mask-test-page-mask-dialog-id-content"
        >
          <div
            className="ajc-flex-row ajc-flex-justify-center ajc-width-hd-inner"
          >
            <div
              className="ajc-flex-column ajc-flex-justify-left ajc-width-dialog-parent"
              id="test-page-mask-dialog-id-root"
            >
              <div
                className="ajc-flex-row ajc-flex-justify-left ajc-dialog-title ajc-width-dialog ajc-text-white ajc-font-default ajc-background-red"
                id="test-page-mask-dialog-id-title"
              >
                <h2
                  className="ajc-font-dialog-title"
                >
                  Test Page Mask Dialog Title
                </h2>
              </div>
              <div
                className="ajc-flex-row ajc-flex-justify-left ajc-dialog-content ajc-width-dialog-content ajc-background-grey-1"
                id="test-page-mask-dialog-id-content-area-root"
              >
                <div
                  className="ajc-flex-column ajc-flex-justify-left ajc-text-black ajc-font-default"
                  id="test-page-mask-dialog-id-content-area"
                >
                  Test Page Mask Dialog Content.
                </div>
              </div>
              <div
                className="ajc-flex-row ajc-flex-justify-center ajc-dialog-footer ajc-width-dialog ajc-font-default ajc-background-red"
                id="test-page-mask-dialog-id-footer"
              >
                <button
                  className="ajc-dialog-footer-button ajc-text-white ajc-menu-item-red"
                  id="save-button"
                  onClick={[Function]}
                  type="submit"
                >
                  Save
                </button>
                <button
                  className="ajc-dialog-footer-button ajc-text-white ajc-menu-item-red"
                  id="cancel-button"
                  onClick={[Function]}
                  type="button"
                >
                  Cancel
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <button
    className="ajc-flex-row ajc-button ajc-button-page-mask-close ajc-text-white ajc-font-default ajc-button-red"
    id="page-mask-test-page-mask-dialog-id-close-button"
    onClick={[Function]}
    title="Click to close"
  >
     X 
  </button>
</div>
`;
